libcalico-go/lib/ipam/ipam.go

```diff
@@ -992,16 +992,13 @@ func (c ipamClient) ReleaseIPs(ctx context.Context, ips ...ReleaseOptions) ([]ne
 		switch ip.Version() {
 		case 4:
 			pool, err = c.blockReaderWriter.getPoolForIP(*ip, v4Pools)
-			if err != nil {
```

Comment:
This is the only diff here that I would prefer to revert - we like to keep error handling immediately after the call that generates the error. It's slightly more verbose but is less prone to unexpected bugs in the future as this code changes.

Reply:
> This is the only diff here that I would prefer to revert - we like to keep error handling immediately after the call that generates the error. It's slightly more verbose but is less prone to unexpected bugs in the future as this code changes.

okay, I've recovered